genie-web/src/main/java/com/netflix/genie/web/agent/launchers/impl/LocalAgentLauncherImpl.java

```diff
@@ -0,0 +1,169 @@
+/*
+ *
+ *  Copyright 2019 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.genie.web.agent.launchers.impl;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.netflix.genie.web.agent.launchers.AgentLauncher;
+import com.netflix.genie.web.dtos.ResolvedJob;
+import com.netflix.genie.web.exceptions.checked.AgentLaunchException;
+import com.netflix.genie.web.properties.LocalAgentLauncherProperties;
+import com.netflix.genie.web.util.ExecutorFactory;
+import io.micrometer.core.instrument.MeterRegistry;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.exec.CommandLine;
+import org.apache.commons.exec.DefaultExecuteResultHandler;
+import org.apache.commons.exec.ExecuteException;
+import org.apache.commons.exec.Executor;
+import org.apache.commons.lang3.SystemUtils;
+
+import javax.validation.Valid;
+import java.io.IOException;
+
+/**
+ * Implementation of {@link AgentLauncher} which launched Agent instances on the local Genie hardware.
+ *
+ * @author tgianos
+ * @since 4.0.0
+ */
+public class LocalAgentLauncherImpl implements AgentLauncher {
+
+    private static final String SETS_ID = "setsid";
```

Comment:
We may move these from the agent into an internal common module, so they can be used here.

Reply:
They're not in the agent codebase. They're in an `agent` package within web which is the only place they'd be used that I can think of